tests/herbie.egg

```diff
@@ -0,0 +1,204 @@
+;; Implements the simplification layer of herbie in egg-smol🫡
+
+;; Bools!
+(datatype B (True) (False))
+(function or (B B) B)
+(rewrite (or (True) x) (True))
+(rewrite (or x (True)) (True))
+(rewrite (or (False) (False)) (False))
+
+;; Operators from FPCore
+(datatype Math
+  ; Ground terms
+  (Num Rational)
+  (Var String)
+
+  ; Constant-folding ops
+  (Add Math Math)
+  (Sub Math Math)
+  (Mul Math Math)
+  (Div Math Math)
+  (Pow Math Math)
+  (Neg Math)
+  (Sqrt Math)
+  (Cbrt Math) ; cube root
+  (Fabs Math)
+  (Ceil Math)
+  (Floor Math)
+  (Round Math)
+  (Log Math))
+
+;; This example has four analyses:
+;; an interval analysis consisting of a hi and lo component,
+;; a constant folding analysis, and a nonzero analysis
+(function hi (Math) Rational :merge (min old new))
+(function lo (Math) Rational :merge (max old new))
+(function evals-to (Math) Rational)
```

Comment:
I'm not sure we need an evals-to function, since you should be able to just match on the `Num` constructor in the query, right?